1-Draft/RFC00XX-PredIctive-IntelliSense.md

```diff
@@ -0,0 +1,346 @@
+
+Users may change the view at the command line using the keybinding F2 or `Set-PSReadLineOption`. The
+`Set-PSReadLineOption` may be stored in the user's profile.
+
+```powershell
+Set-PSReadLineOption -PredictionViewStyle ListView
+```
+
+### Change the Prediction Color for Accessibility
+
+By default, predictions appear on the same line the user is typing in a different customizable
+color. To support accessibility needs, the prediction color is settable in the shell or user's
+profile.
+
+The default light-grey prediction text color;
+
+```powershell
+Set-PSReadLineOption -Colors @{ InLinePrediction = "$([char]0x1b)[38;5;238m"}
+```
+
+Starting with PSReadLine 2.1.0-RC1, `Set-PSReadLineOption -Colors` array will include
+**InLinePrediction** to set the color of predictive text for **InLineView**.
+
+Beginning with PSReadLine 2.2.0-beta1, `Set-PSReadLineOption -Colors` array will include;
+
+- **ListPredictionColor** to set color for '>' and source names in **ListView**
+- **ListPredictionSelectionColor** to set color for the highlighted selection in **ListView**
+
+Examples of user changing prediction color:
+
+![pi-color](./media/pi-color.png)
+![pi-color](./media/pi-color2.png)
+![pi-color](./media/pi-color3-background.png)
+
+```powershell
+Set-PSReadLineOption -Colors @{ InLinePrediction = '#8A0303'}
+Set-PSReadLineOption -Colors @{ InLinePrediction = '#2F7004'}
+Set-PSReadLineOption -Colors @{ InLinePrediction = "$([char]0x1b)[36;7;238m"}
+```
+
+Multiple types of color code values are supported in PSReadLine. For more information see
+Set-PSReadLineOption in
+[PSReadLine](https://docs.microsoft.com/en-us/powershell/module/PSReadline/Set-PSReadlineOption?view=powershell-7)
+
+Examples of different color code values:
+
+```powershell
+ # Use a ConsoleColor enum
+Set-PSReadLineOption -Colors @{ InLinePrediction = 'DarkRed'}
+Set-PSReadLineOption -Colors @{ InLinePrediction = [ConsoleColor]::DarkRed}
+
+ # 24 bit color escape sequence
+Set-PSReadLineOption -Colors @{ InLinePrediction = "$([char]0x1b)[38;5;100m"}
+
+ # RGB value
+Set-PSReadLineOption -Colors @{ InLinePrediction = "#8181f7"}
+```
+
+### Key Bindings for Predictions
+
+Key bindings control cursor movement and additional features within the prediction. To support users
+running Predictive IntelliSense on multiple platforms, key bindings are user-settable in the shell
+and a user’s PowerShell profile.
+
+Example of setting a key binding in the shell or user’s profile:
+
+```powershell
+Set-PSReadLineOption -HistorySearchCursorMovesToEnd
+Set-PSReadLineKeyHandler -Key UpArrow -Function HistorySearchBackward
+Set-PSReadLineKeyHandler -Key DownArrow -Function HistorySearchForward
+```
+
+List of suggested key bindings defined in **SamplePSReadLineProfile.ps1**
+
+Key Binding   | PSReadLine Function    | Description
+------------- | -------------------    | -----------
+Up Arrow	    | HistorySearchBackward  | Move backward through history
```

Comment:
Can you line up the vertical bars?

Reply:
I adjusted again - let's see if they stay when I update ;)

Reply:
@SteveL-MSFT Turns out that they display correctly if you view the file, but in the raw view they get misaligned.  The original file has them all straight.